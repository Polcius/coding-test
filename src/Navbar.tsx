import React, { FunctionComponent, useState } from "react";
import styled from "styled-components";

const NavHeader = styled.nav`
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: #def1f4;
  margin-bottom: 0.5rem;
  * {
    margin: 0.5rem 1rem;
  }
`;
const ProductDetails = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  .product-name {
    font-weight: bold;
  }
  .product-price {
    font-size: 1.5rem;
    font-weight: bold;
  }
`;

// type CardProps = {
//   image: string;
//   name: string;
//   price: string;
//   isSale: boolean;
//   isExclusive: boolean;
// }

type NavbarProps = {
  onChange: (e: React.ChangeEvent<HTMLSelectElement>) => void;
};

const Navbar: FunctionComponent<NavbarProps> = ({ onChange }) => {
  const sizes = ["XS", "S", "M", "L", "XL"];
  return (
    <NavHeader>
      <h1>Women's tops</h1>
      <select onChange={onChange} name="sizes" id="size-select">
        <option value="">Filter by size:</option>
        {sizes.map(size => (
          <option value={size} key={size}>
            {size}
          </option>
        ))}
      </select>
    </NavHeader>
  );
};

export default Navbar;
