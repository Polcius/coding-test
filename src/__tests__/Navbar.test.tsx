import React from "react";
import {
  render,
  cleanup,
  waitForElement,
  fireEvent
} from "@testing-library/react";
import Navbar from "../Navbar";
afterEach(() => {
  cleanup();
  console.error.mockClear();
});

console.error = jest.fn();

test("<Navbar />", async () => {
  const text = "my text";

  const { getByTestId, getAllByTestId, queryByTestId, getByText } = render(
    <Navbar headingText={text} onChange={jest.fn()} />
  );

  expect(getByTestId("navbar-text")).toBeTruthy();
  expect(getByTestId("navbar-text")).toHaveTextContent(text);
  expect(getByText(/filter by size:/i)).toBeInTheDocument();
  expect(getByTestId("size-select")).toMatchSnapshot();
});

// test("<ProductGrid /> API fail", async () => {
//   fetch.mockReject(new Error("fake error message"));

//   const { getByTestId, queryByTestId } = render(<ProductGrid sizeFilter="" />);
//   expect(getByTestId("loading")).toBeTruthy();
//   await waitForElement(() => getByTestId("wrapper"));
//   expect(queryByTestId("loading")).toBeFalsy();
//   expect(queryByTestId("error")).toBeTruthy();
// });
