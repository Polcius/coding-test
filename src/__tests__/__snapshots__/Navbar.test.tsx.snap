// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> 1`] = `
<select
  data-testid="size-select"
  id="size-select"
  name="sizes"
>
  <option
    value=""
  >
    Filter by size:
  </option>
  <option
    value="XS"
  >
    XS
  </option>
  <option
    value="S"
  >
    S
  </option>
  <option
    value="M"
  >
    M
  </option>
  <option
    value="L"
  >
    L
  </option>
  <option
    value="XL"
  >
    XL
  </option>
</select>
`;
